lib/OpenQA/WebAPI/Controller/API/V1/Table.pm

```diff
@@ -320,4 +284,48 @@ sub destroy {
     $self->render(json => {result => int($ret)});
 }
 
+=over 4
+
+=item prepare_settings()
+
+Internal method to prepare settings when add or update admin table.
+Use by both B<create()> and B<update()> method.
+
+=back
+
+=cut
+
+sub prepare_settings {
+    my ($self, $table, $entry) = @_;
+    my $validation = $self->validation;
+
+    for my $par (@{$tables{$table}->{required}}) {
+        $validation->required($par);
+        if (!defined $validation->param($par)) {
+            next;
+        }
+        $entry->{$par} = trim $validation->param($par);
+    }
+
+    my $error_message;
```

Comment:
This doesn't seem to be used anymore

Reply:
if I remove it, the following return should be modified to `return (undef, \@settings, \@keys)`, is it ok?